### ISSUE TITLE
import map on transpileOnly

### ISSUE THREAD

**Issue by newtack** [denoland/deno#6831]
transpileOnly fails because it does not use the same import map that the running script uses.

Can we either have the currently active import map be used by transpileOnly or have an option to pass it in?

**Comment by kitsonk** [comment]
`Deno.transpileOnly` does not resolve modules at all. You have to supply it sources. An import map makes no sense in that context.

**Comment by qballer** [comment]
@newtack, I haven't read the Deno `transpileOnly` source code by I'm assuming it leans on the `transpileOnly` API provided by the typescript compiler. As such, it doesn't crawl the import statement which would explain @kitsonk comment about not resolving modules.

Can you share some details as to why you are using this API and not `run` or `build`?

**Comment by newtack** [comment]
I am building a dev server and transpile the client side Typescript files on-demand. Something like this:
https://github.com/zhmushan/dev_server/blob/master/mod.ts
line 137.

Since I use transpileOnly I really don't want to provide other sources since I just want to transpile this file. But when I did this in a file with an import like this

import {Counter} from "p/ui/Counter.tsx;

and p is defined in import_map.json as
"p/": "./platform/",

then I got an error during transpileOnly complaining that it must be a relative import. For now I worked around it by manually replacing "p/" with "../../../platform/", but I would like to not use this hack.
